// arcgis/dist/node/item/layers.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = layerMethods;
var end = require('../lib/end');

function layerMethods(i, arcgis) {
  //  Will require stronger Create voodoo
  //  i.duplicate = function (cb) {
  //    var duplicatePromise =
  //    return end(duplicatePromise, cb)
  //  }
  i.data = function () {
    var layer = arguments.length <= 0 || arguments[0] === undefined ? 0 : arguments[0];
    var cb = arguments[1];

    var dataPromise = arcgis.request({
      url: '/query',
      form: { where: '1=1' },
      rootUrl: i.url + '/' + layer
    });
    return end(dataPromise, cb);
  };
  // Doesn't work ???
  i.exportLayer = function () {
    var format = arguments.length <= 0 || arguments[0] === undefined ? 'GeoJSON' : arguments[0];
    var cb = arguments[1];

    var exportOptions = {
      itemId: i.id,
      exportFormat: format
    };
    var exportPromise = arcgis.request({
      url: 'content/users/' + i.owner + '/export',
      form: exportOptions,
      post: true
    });
    return end(exportPromise, cb);
  };
  //  Don't know how this works yet
  //  i.generateTiles = function (cb) {
  //    var tilesPromise =
  //    return end(tilesPromise, cb)
  //  }
  //  This is complicated?
  //  i.usage = function (cb) {
  //    var usagePromise =
  //    return end(usagePromise, cb)
  //  }
  return i;
}
