// arcgis/dist/node/user/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = User;
var end = require('../lib/end');

function User(arcgis) {
  function get(username) {
    return arcgis.request({ url: 'community/users/' + username });
  }

  return function getUser(username, cb) {
    var promise = get(username).then(function (u) {
      u.get = get;
      u.update = function (options, cb) {
        var updatePromise = arcgis.request({
          url: 'community/users/' + username + '/update',
          form: options,
          post: true
        }).then(function (results) {
          return u.get();
        });
        return end(updatePromise, cb);
      };
      u.delete = function (cb) {
        return end(arcgis.request({
          url: 'community/users/' + username + '/delete',
          form: {},
          post: true
        }), cb);
      };
      u.content = function () {
        var _ref = arguments.length <= 0 || arguments[0] === undefined ? {} : arguments[0];

        var _ref$num = _ref.num;
        var num = _ref$num === undefined ? 100 : _ref$num;
        var _ref$page = _ref.page;
        var page = _ref$page === undefined ? 0 : _ref$page;
        var _ref$sort = _ref.sort;
        var sort = _ref$sort === undefined ? 'created' : _ref$sort;
        var _ref$order = _ref.order;
        var order = _ref$order === undefined ? 'desc' : _ref$order;
        var cb = arguments[1];

        return end(arcgis.search({
          queryString: '"" owner:' + username,
          page: page,
          sort: sort,
          order: order
        }), cb);
      };
      u.favorites = function (options, cb) {
        var favoritesPromise = arcgis.group(u.favGroupId).then(function (g) {
          return g.content(options);
        });
        return end(favoritesPromise, cb);
      };
      u.tags = function (cb) {
        return end(arcgis.request({
          url: 'community/users/' + username + '/tags'
        }), cb);
      };
      u.enabled = function (bool, cb) {
        if (bool === false) {
          return end(arcgis.request({
            url: 'community/users/' + username + '/disable',
            form: {},
            post: true
          }), cb);
        } else {
          return end(arcgis.request({
            url: 'community/users/' + username + '/enable',
            form: {},
            post: true
          }), cb);
        }
      };
      return u;
    });
    return end(promise, cb);
  };
}
