// arcgis/dist/node/group/create.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CreateGroup;

var _end = require('../lib/end');

var _end2 = _interopRequireDefault(_end);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function CreateGroup(arcgis) {
  return function create() {
    var _ref = arguments.length <= 0 || arguments[0] === undefined ? {} : arguments[0];

    var title = _ref.title;
    var description = _ref.description;
    var summary = _ref.summary;
    var tags = _ref.tags;
    var _ref$access = _ref.access;
    var access = _ref$access === undefined ? 'private' : _ref$access;
    var _ref$isViewOnly = _ref.isViewOnly;
    var isViewOnly = _ref$isViewOnly === undefined ? false : _ref$isViewOnly;
    var _ref$isInvitationOnly = _ref.isInvitationOnly;
    var isInvitationOnly = _ref$isInvitationOnly === undefined ? false : _ref$isInvitationOnly;
    var cb = arguments[1];

    var options = {
      title: title,
      description: description,
      snippet: summary,
      tags: tags.join(','),
      access: access,
      isViewOnly: isViewOnly,
      isInvitationOnly: isInvitationOnly
    };
    var createPromise = arcgis.request({
      url: 'community/createGroup',
      form: options,
      post: true
    }).then(function (confirmation) {
      return confirmation.group;
    });

    return (0, _end2.default)(createPromise, cb);
  };
}
